R/data.table.R

```diff
@@ -2962,25 +2969,13 @@ setDT = function(x, keep.rownames=FALSE, key=NULL, check.names=FALSE) {
   if (is.name(name)) {
     name = as.character(name)
     assign(name, x, parent.frame(), inherits=TRUE)
-  } else if (.is_simple_extraction(name)) {
```

Comment:
please be aware that the underlying code has changed since what was permalinked in #6702. In particular note this `.is_simple_extraction()` helper function which was not present then. Please adapt.